### ISSUE TITLE
Request to update mermaid version

### ISSUE THREAD

**Issue by CHAOSdonut** [digitalbazaar/respec-mermaid#5]
The current version of the project uses v9.0.0, that version appears to have problems with linking to subgraphs that are fixed in the latest version: v9.3.0

Could the CDN be updated to use the latests version of mermaid?

**Comment by jandrieu** [comment]
Yes, please. =)

@msporny

**Comment by msporny** [comment]
Yep, let me see if I can merge PR #6 and get a new version out there.

**Comment by msporny** [comment]
I'm using jsdelivr, which requires tags IIRC. 

I failed to do a new release, my webpack build is hanging on both node v18 and v20 and I don't have time to debug it this week. That's going to hold up the release. I hear you know a thing or two about webpack, @davidlehn... I even heard that you might be the webpack messiah. That's what the word on the street is, anyways.

Any chance you could take a shot at a release that is jsdelivr'able (or anything equivalent). Note the Usage line here:

https://github.com/digitalbazaar/respec-mermaid#usage

Ultimately, that's what we have to get to.

**Comment by davidlehn** [comment]
I've created a patch to help with this.  I've got a couple questions though.

@jandrieu:
- The mermaid version bump required an API change to function.

@msporny:
- I bumped the cli version, that might help your webpack woes.  I had a similar issue in another project.  Old webpack has some issues with new nodes.
- The output has more than one "dist/main.js" file and outputs a slew of other files.  I'm not sure if it's dynamically loading those based on diagram type or what.  Maybe they are all needed now?  Or were those not checked in before?
- The output is also in `development` mode and rather large.  Is that ok for this use case?
- jsdelivr supports npm packages too, so that could be used.  Would need to point people at a different URL is all.  We could use the digitalbazaar tooling for releases for this.  Thoughts?

**Comment by msporny** [comment]
@davidlehn wrote: 
> * I bumped the cli version, that might help your webpack woes.  I had a similar issue in another project.  Old webpack has some issues with new nodes.

Seems the word on the street about you being the webpack messiah were accurate.

> * The output has more than only the "dist/main.js" file and outputs a slew of other files.  I'm not sure if it's dynamically loading those based on diagram type or what.  Maybe they are all needed now?  Or were those not checked in before?

I have no idea, those new files are ... new. There used to only be one dist/main.js created. We might try removing the extraneous files and seeing what would happen. I was hesitant to check in any auto-generated file -- the only place the auto-generated files exist is in the tagged versions, which are orphan off-shoots of the main branch.

> * The output is also in `development` mode and rather large.  Is that ok for this use case?

No, we want it in production mode, IMHO.

> * jsdelivr supports npm packages too, so that could be used.  Would need to point people at a different URL is all.  We could use the digitalbazaar tooling for releases for this.  Thoughts?

Pointing people to different URLs for new versions is fine. Let's try to use DB tooling for releases and see where this takes us. Remember that we're going to hand this over to W3C for maintenance (same maintainers -- us -- but, different org).  Let's shift over to npm packages if we can... the old process is tedious and error-prone.

Thanks for looking into this @davidlehn ! :)

**Comment by jandrieu** [comment]
Also, Please check the function of the output in the new main.js.

I dropped that in directly and my respec mermaid failed. 

Oddly, I didn't have any problems with `npm run build` after updating the package.json. Just that the figure is failing with an error about not being able to CreateElementNS.

I'm looking into that now.

**Comment by jandrieu** [comment]
FWIW:
```
TypeError: Cannot read properties of undefined (reading 'createElementNS')
    at Function.eval (creator.js:16:20)
    at Function.eval (append.js:11:36)
    at Selection.__WEBPACK_DEFAULT_EXPORT__ [as select] (select.js:15:50)
    at Selection.__WEBPACK_DEFAULT_EXPORT__ [as append] (append.js:10:15)
    at appendDivSvgG (mermaid-934d9bea.js:5977:35)
    at Object.render$1 [as render] (mermaid-934d9bea.js:6028:5)
    at createFigures (index.js:48:63)
    at pre-process.js:28:24
    at Array.map (<anonymous>)
    at pre-process.js:26:8
```

**Comment by davidlehn** [comment]
@jandrieu: Are you using https://github.com/digitalbazaar/respec-mermaid/pull/7?  I think with the API fix there, tool update, rebuild, and force reloads (and/or shutting caching off), the test page will run.

**Comment by msporny** [comment]
@jandrieu wrote:
>  Just that the figure is failing with an error about not being able to CreateElementNS.

Yep, same issue that I hit. Tooling update the @davidlehn mentioned might fix this.

**Comment by msporny** [comment]
respec-mermaid v1.1.0 has been released, which includes Mermaid v10.6.1. Closing.

**Comment by jandrieu** [comment]
I'll check it out.
